### ISSUE TITLE
Code splitting

### ISSUE THREAD

**Issue by louischan-oursky** [authgear/authgear-server#2058]
Goals

- Code splitting
- No longer need to manually declare dependencies in `static_assets.go`
- Manage third dependencies with `authui/package.json`
- Rebuild `authui` does not require restart of server to serve newer assets
- Do NOT need to change our code to use dynamic import

Idea

- Introduce GeneratedAssetDescriptor.
- It reads a generated manifest JSON file to serve ALL generated assets
- It does NOT read the manifest every time.
- It watches the change of the manifest, update its state

**Comment by louischan-oursky** [comment]
Example of including classic script and CSS

```html
<!DOCTYPE html>
<html>
<head>
	<link rel="stylesheet" href="npm:intl-tel-input/build/css/intlTelInput.css">
	<link rel="stylesheet" href="npm:@tabler/icons/iconfont/tabler-icons.min.css">
	<script src="npm:intl-tel-input/build/js/intlTelInput.js"></script>
	<script src="npm:intl-tel-input/build/js/utils.js"></script>
	<script type="module" src="./pageA.ts"></script>
</head>
</html>

```

**Comment by louischan-oursky** [comment]
Problems we have encountered

## Parcel requires either `import()` or multiple page to enable code splitting

Our solution is to create a dummy page that refers a dummy script. The dummy script contains `import()` calls to force code splitting on third party modules

```html
<!DOCTYPE html>
<html>
  <head>
    <script type="module" src="./dummy.ts"></script>
  </head>
</html>
```

```typescript
import("library-a");
import("library-b");
```

```sh
$ parcel build ./src/*.html
```

## FOUC after code splitting

Our color scheme is class-based so it requires JavaScript to work. The color scheme JavaScript was initially bundled together with other non-rendering essential code. It runs only after every script has been loaded. However, there is a gap between all blocking CSS have been loaded and all deferred scripts have been fetched. Once blocking CSS have been loaded, the page renders immediately. At that time, the color scheme JavaScript did not kick in yet, thus the page is rendered in light. After all deferred scripts have been executed, the page is rendered in dark. This light to dark change is the FOUC. We solve this by extracting the color scheme JavaScript from the main bundle, making it a blocking classic script. So it is executed BEFORE the page is rendered.
